llvm/lib/Target/EraVM/EraVMTargetMachine.cpp

```diff
@@ -292,8 +292,15 @@ void EraVMPassConfig::addPreRegAlloc() {
     addPass(createEraVMOptimizeSelectPreRAPass());
     addPass(createEraVMHoistFlagSettingPass());
     addPass(&LiveVariablesID);
-    addPass(createEraVMTieSelectOperandsPass());
   }
+  // Since some conditional codes like overflow LT hasn't reversal version (the
+  // GE isn't suitable according to spec), we have to rely on this pass to tie
+  // the output with the second input operand (the false part), otherwise if we
+  // get below IR, we will be unable to expand it correctly:
+  //   $r1 = SELrrr killed $r1, killed $r6, COND_OF, implicit $flags, implicit
+  //   $r1(tied-def 0)
+  // Therefor we enable it for all optimization level.
+  addPass(createEraVMTieSelectOperandsPass());
```

Comment:
Please don't make this pass available for all optimization levels. This is pass to optimize select patterns and it is not pass where we want to resolve functional problems. This problem is present in all SELr variants, not only SELrrr. Please check the explanation how this should be addressed.

Reply:
Reverted this change now.